tests/unit/package_managers/test_gomod.py

```diff
@@ -1566,3 +1567,94 @@ def test_dep_replacements(
         fetch_gomod_source(request)
         expected_calls = [mock.call(request.source_dir / pkg.path, request) for pkg in packages]
         mock_resolve_gomod.assert_has_calls(expected_calls, any_order=True)
+
+
+@pytest.mark.parametrize(
+    "pkg_config",
+    (
+        [],
+        [
+            {
+                "type": "gomod",
+                "path": ".",
+                "name": "github.com/my-org/my-repo",
+                "version": "1.0.0",
+                "dependencies": [
+                    {
+                        "name": "golang.org/x/net",
+                        "type": "gomod",
+                        "version": "v0.0.0-20190311183353-d8887717615a",
+                    }
+                ],
+            },
+        ],
+        [
+            {
+                "type": "gomod",
+                "path": ".",
+                "name": "github.com/my-org/my-repo",
+                "version": "1.0.0",
+                "dependencies": [],
+            },
+            {
+                "type": "gomod",
+                "path": "path",
+                "name": "github.com/my-org/my-repo/path",
+                "version": "1.0.0",
+                "dependencies": [],
+            },
+        ],
+    ),
+)
+@mock.patch("cachi2.core.package_managers.gomod._find_missing_gomod_files")
+@mock.patch("cachi2.core.package_managers.gomod._resolve_gomod")
+@mock.patch("cachi2.core.models.input.Request")
+def test_fetch_gomod_source(
+    mock_request, mock_resolve_gomod, mock_find_missing_gomod_files, pkg_config, env_variables
+):
+    packages = [PackageInput(path=pkg["path"], type="gomod") for pkg in pkg_config]
+
+    request = mock_request()
+    request.packages = packages
+    request.source_dir = Path(".")
+    request.dep_replacements = []
+
+    def resolve_gomod_mocked(path, request):
+        package = list(filter(lambda p: p["path"] == str(path), pkg_config))[0]
+
+        return {
+            "module": {"type": "gomod", "name": package["name"], "version": package["version"]},
+            "module_deps": [*package["dependencies"]],
+            "packages": [
+                {
+                    "pkg": {
+                        "type": "go-package",
+                        "name": package["name"],
+                        "version": package["version"],
+                    },
+                    "pkg_deps": [*package["dependencies"]],
+                }
+            ],
+        }
```

Comment:
By any chance, do we have the expected output of `resolve_gomod()` defined somewhere as a constant or fixture? If yes, could we use that instead?

If not, this seems fine, maybe just fix the type of `pkg_deps` (should be "go-package")

Reply:
We don't have a resolve_gomod_output fixture. We could turn this into one, but it would require the same fixture override trick we did with gomod_request, and it would not be used anywhere else for now.

Think it's still worth it?

Reply:
Nah, probably not then